src/main/java/core/basesyntax/HelloWorld.java

```diff
@@ -4,5 +4,18 @@
  * Feel free to remove this class and create your own.
  */
 public class HelloWorld {
+    private static final int ARRAY_SIZE = 6;
 
+    public static void main(String[] args) {
+        Figure[] figures = new Figure[ARRAY_SIZE];
+        FigureSupplier figureSupplier = new FigureSupplier();
+        for (int i = 0; i < ARRAY_SIZE; i++) {
+            if (i < ARRAY_SIZE / 2) {
+                figures[i] = figureSupplier.getRandomFigure();
+            } else {
+                figures[i] = figureSupplier.getDefaultFigure();
+            }
+            figures[i].draw();
+        }
```

Comment:
The loop is correctly iterating over the array and populating it with figures. It's good that you have avoided creating a new `FigureSupplier` inside the loop, which is in line with the checklist.